python/pyarrow/_fs.pyx

```diff
@@ -484,10 +484,17 @@ cdef class FileSystem:
         ----------
         path : str
             The path of the directory to be deleted.
+        accept_root_dir : boolean, default False
+            Allow deleting the root directory's contents
+            (if path is empty or "/")
         """
         cdef c_string directory = _path_as_bytes(path)
-        with nogil:
-            check_status(self.fs.DeleteDirContents(directory))
+        if accept_root_dir and directory in (b"", b"/"):
```

Comment:
IsEmptyPath is true for any number of slashes, shouldn't we allow that here as well?

Reply:
Sure.